From 2316632e8f3eefc21bc4f9cb97be4603b4c14719 Mon Sep 17 00:00:00 2001
From: Mark Hatle <mark.hatle@amd.com>
Date: Thu, 28 Sep 2023 12:24:04 -0600
Subject: [PATCH] Current gcc requires cstdint for C types

Add #include <cstdint> to resolve the issues similar to the following:

  module/module_structs.h:259:3: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
  module/module_structs.h:260:3: error: 'uint16_t' does not name a type

Signed-off-by: Mark Hatle <mark.hatle@amd.com>
---
 exe_omx/encoder/EncCmdMngr.h | 1 +
 module/module_structs.h      | 1 +
 utility/processor_fifo.h     | 1 +
 3 files changed, 3 insertions(+)

diff --git a/exe_omx/encoder/EncCmdMngr.h b/exe_omx/encoder/EncCmdMngr.h
index 6dacd68..cd3d0a6 100644
--- a/exe_omx/encoder/EncCmdMngr.h
+++ b/exe_omx/encoder/EncCmdMngr.h
@@ -7,6 +7,7 @@
 #include <iostream>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "ICommandsSender.h"
 
diff --git a/module/module_structs.h b/module/module_structs.h
index 7151b86..37ff8ac 100644
--- a/module/module_structs.h
+++ b/module/module_structs.h
@@ -6,6 +6,7 @@
 #include "module_enums.h"
 #include <string>
 #include <vector>
+#include <cstdint>
 
 template<typename T>
 struct InputOutput
diff --git a/utility/processor_fifo.h b/utility/processor_fifo.h
index 1c62ba4..3c9cd86 100644
--- a/utility/processor_fifo.h
+++ b/utility/processor_fifo.h
@@ -6,6 +6,7 @@
 #include <utility/locked_queue.h>
 #include <thread>
 #include <functional>
+#include <string>
 
 #if defined __linux__
 #include <sys/prctl.h>
-- 
2.34.1
